presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -862,24 +868,46 @@ public static Slice stEndPoint(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     }
 
     @SqlNullable
-    @Description("Returns an array of points in a linestring")
+    @Description("Returns an array of points in a geometry")
     @ScalarFunction("ST_Points")
     @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
     public static Block stPoints(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
         Geometry geometry = deserialize(input);
-        validateType("ST_Points", geometry, EnumSet.of(LINE_STRING));
+        validateType("ST_Points", geometry, VALID_TYPES_FOR_ST_POINTS);
```

Comment:
Since this is all current geometries, and the method would work for any additional geometry that JTS supports, we can remove this (and the associated bit in lines 150-152.

Reply:
Sounds great. Removing this check.